src/backend/distributed/executor/executor.c

```diff
@@ -2234,6 +2341,35 @@ WorkerPoolFailed(WorkerPool *workerPool)
 
 	/* we do not want more connections in this pool */
 	workerPool->readyTaskCount = 0;
+	workerPool->failed = true;
+
+	/*
+	 * When each task is forced to be executed over a single connection,
+	 * we should consider the worker failover (e.g., replication > 1) and
+	 * restart the connection timeout.
```

Comment:
Maybe elaborate on why, e.g.

The reason is that when replication factor is > 1 and we are performing a SELECT, then we only establish connections for the specific placements that we will read from. However, when a worker pool fails, we will need to establish multiple new connection to other workers and the query can only succeed if all those connections are established.